src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -3327,13 +3318,14 @@ protected override bool HasChildItems(string path)
             path = NormalizePath(path);
 
             // First check to see if it is a directory
-
             try
             {
                 DirectoryInfo directory = new DirectoryInfo(path);
 
                 // If the above didn't throw an exception, check to
-                // see if it contains any directories
+                // see if we should proceed and if it contains any children
```

Comment:
This is a bit surprising.  What case does this cover where an exception is not thrown but Attributes is not Directory?  Please add a comment to clarify.